packages/gatsby/src/utils/sanitize-node.ts

```diff
@@ -20,15 +68,15 @@ const sanitizeNode = (data, isNode = true, path = new Set()) => {
         returnData[key] = o
         return
       }
-      returnData[key] = sanitizeNode(o, false, path)
+      returnData[key] = sanitizeNode(o as data, false, path)
```

Comment:
Can we change this loop to a `Object.entries(data).forEach`? That should allow TS to pick up the right types and we shouldn't need to cast.

Reply:
`Object.entries` hardcodes the first value `o` as string, making tests fail (Symbol/string issues)